File: 정건우/6주차/260206.java
//https://www.acmicpc.net/problem/13234
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ_B2_13234_GeorgeBoole {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        boolean v1 = st.nextToken().equals("true");
        String option = st.nextToken();
        boolean v2 = st.nextToken().equals("true");

        if(option.equals("AND")) {
            System.out.println(v1 && v2);
        } else {
            System.out.println(v1 || v2);
        }

    }
}

File: 심수연/6주차/260206.py
# https://www.acmicpc.net/problem/21918

import sys
input = sys.stdin.readline

N, M = map(int, input().split())
bulb = list(map(int, input().split()))

for i in range(M):
  a, b, c = map(int, input().split())
  if a == 1:
    bulb[b - 1] = c
  elif a == 2:
    for j in range(b - 1, c):
      bulb[j] = int(not bulb[j])
  elif a == 3:
    for j in range(b - 1, c):
      bulb[j] = 0
  else:
    for j in range(b - 1, c):
      bulb[j] = 1

print(*bulb)